pkg/controller/elasticsearch/pdb/reconcile.go

```diff
@@ -116,11 +116,13 @@ func expectedPDB(es esv1beta1.Elasticsearch, statefulSets sset.StatefulSetList,
 }
 
 // buildPDBSpec returns a PDBSpec computed from the current StatefulSets,
-// considering the cluster health and topology.
+// considering the cluster health and topology, when cluster is high available.
+// it sets MinAvailable to 0 when nodecount is 1 (not high available)
+//   when the PDB is 1, and the amount of pods is 1, Kubernetes will not be able to reboot the node.
```

Comment:
I think I would remove the newly added comment here since the logic is now moved to another function.
The content of `allowedDisruptions()` is already very well documented and self-explanatory. WDYT?

Reply:
You are totally right, reverted to the previous code.
Apologies for the hasty job.

Reply:
No worries!